catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/UserResourceTest.java

```diff
@@ -109,20 +109,49 @@ void post_userWithDefaultRole(TestInfo test) throws IOException {
 
     // Given a default role has been set, when a user is created without any roles, then the default role should be
     // assigned.
-    CreateUser create = createRequest(test, 2);
+    CreateUser create = createRequest(test, 1);
     createUserAndCheckRoles(create, Arrays.asList(defaultRoleId));
 
     // Given a default role has been set, when a user is created with a non default role, then the default role should
     // be assigned along with the non default role.
-    create = createRequest(test, 3).withRoles(List.of(nonDefaultRoleId));
+    create = createRequest(test, 2).withRoles(List.of(nonDefaultRoleId));
     createUserAndCheckRoles(create, Arrays.asList(nonDefaultRoleId, defaultRoleId));
 
     // Given a default role has been set, when a user is created with both default and non-default role, then both
     // roles should be assigned.
-    create = createRequest(test, 4).withRoles(List.of(nonDefaultRoleId, defaultRoleId));
+    create = createRequest(test, 3).withRoles(List.of(nonDefaultRoleId, defaultRoleId));
     createUserAndCheckRoles(create, Arrays.asList(nonDefaultRoleId, defaultRoleId));
   }
 
+  @Test
+  void post_userWithTeamDefaultRole(TestInfo test) throws HttpResponseException {
+    RoleResourceTest roleResourceTest = new RoleResourceTest();
+    List<Role> roles = roleResourceTest.listEntities(Collections.emptyMap(), ADMIN_AUTH_HEADERS).getData();
+    UUID defaultRoleId =
+        roles.stream().filter(Role::getDefault).findAny().orElseThrow().getId(); // DataConsumer is default role.
+    Role role1 = roleResourceTest.createEntity(roleResourceTest.createRequest(test, 1), ADMIN_AUTH_HEADERS);
+    Role role2 = roleResourceTest.createEntity(roleResourceTest.createRequest(test, 2), ADMIN_AUTH_HEADERS);
+
+    TeamResourceTest teamResourceTest = new TeamResourceTest();
+    Team team1 =
+        teamResourceTest.createEntity(
+            teamResourceTest.createRequest(test, 1).withDefaultRoles(List.of(role1.getId())), ADMIN_AUTH_HEADERS);
+    Team team2 =
+        teamResourceTest.createEntity(
+            teamResourceTest.createRequest(test, 2).withDefaultRoles(List.of(role1.getId(), role2.getId())),
+            ADMIN_AUTH_HEADERS);
+
+    // Given team1 with role1 set as default, when a user is created with team1, then the user should have the global
+    // default role and role1 assigned.
+    CreateUser create = createRequest(test, 1).withTeams(List.of(team1.getId()));
+    createUserAndCheckRoles(create, Arrays.asList(defaultRoleId, role1.getId()));
+
+    // Given team1 with role1 set as default and team2 with role1 and role2 set as default, when a user is created
+    // with team1 and team2, then the user should have the global default role, role1 and role2 assigned.
+    create = createRequest(test, 2).withTeams(List.of(team1.getId(), team2.getId()));
+    createUserAndCheckRoles(create, Arrays.asList(defaultRoleId, role1.getId(), role2.getId()));
```

Comment:
Should probably add a test to showcase removal of a role when user changes teams